chromium_src/components/permissions/permission_request_manager.cc

```diff
@@ -0,0 +1,96 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <string>
+
+#include "brave/components/brave_wallet/common/buildflags/buildflags.h"
+#include "components/permissions/permission_request.h"
+
+#if BUILDFLAG(BRAVE_WALLET_ENABLED)
+#include "brave/components/brave_wallet/browser/ethereum_permission_utils.h"
+
+namespace permissions {
+namespace {
+bool BraveShouldGroupRequests(PermissionRequest* a, PermissionRequest* b);
+}  // namespace
+}  // namespace permissions
+
+#define BRAVE_SHOULD_GROUP_REQUESTS   \
+  if (BraveShouldGroupRequests(a, b)) \
+    return true;
+#else
+#define BRAVE_SHOULD_GROUP_REQUESTS
+#endif  // BUILDFLAG(BRAVE_WALLET_ENABLED)
+
+#include "../../../../components/permissions/permission_request_manager.cc"
+#undef BRAVE_SHOULD_GROUP_REQUESTS
+
+#if BUILDFLAG(BRAVE_WALLET_ENABLED)
+namespace permissions {
+
+namespace {
+
+bool BraveShouldGroupRequests(PermissionRequest* a, PermissionRequest* b) {
```

Comment:
why are we grouping these? It seems like grouping is currently restricted to only mic/camera and notifications, storage access, geo, etc... are all ungrouped

Reply:
When requesting permission for one origin, we will have multiple sub-requests for each account address, we're grouping them together to be handled in one prompt. This is only group when their original requesting_origin is the same (for those sub-requests).

Reply:
but what if I only want to allow some account addresses for an origin? Isn't this all or nothing then?

Reply:
Please see `PermissionRequestManager::AcceptEthereumPermissionRequests`, we checked the accounts user selected and call `PermissionGrantedIncludingDuplicates` if a sub-request's account is selected, otherwise cancel is called.
https://github.com/brave/brave-core/pull/9443/files#diff-1ef0a50093247f1d6161497b88bb559a99bed678357d97c841afbb233baeba09R65
It does make things tricky that Chromium currently don't expect multiple sub-requests have different decisions, so
one unresolved issue here in that exact function is we can only call `FinalizeCurrentRequests` with one single decision. 
Currently seems no harm yet because it only for like things like histogram for now, but definitely something to look into, wonder if you have some thoughts on that one?